### PR TITLE
Elm implementation of challenges

### PR DESCRIPTION
**Pull request by wolfadex**

### CODE REVIEW

File: challenges/elm/src/Minimal.elm [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+{-| #2
+No way to shorten while keeping the readability.
+-}
+bookInfo : JsonValue -> JsonValue -> Result String String
+bookInfo book catalogData =
+    Result.map3
+        (\title isbn authors ->
+            [ ( "title", Json.Encode.string title )
+            , ( "isbn", Json.Encode.string isbn )
```
Comment (viebel) [inline challenges/elm/src/Minimal.elm:26]: Is it mandatory to include json serialization in the "business logic"
would it make sense to have `bookInfo : JsonValue -> JsonValue -> JsonValue` and a generic function `stringify: JsonValue -> String)`?

File: challenges/elm/src/Detailed.elm [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+
+{-| #1 Retrieve a piece of information
+
+In Elm we return a `Result error value` because we're required to handle failure cases.
+
+-}
+getProperty : List String -> JsonValue -> Result String JsonValue
+getProperty keys obj =
+    Json.Value.getIn keys obj
```
Comment (viebel) [inline challenges/elm/src/Detailed.elm:17]: It's cool that json-value provides a `getIn` method!

File: challenges/elm/src/Detailed.elm [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+                                namesJson
+                        )
+            )
+
+
+{-| #3 Add a piece of information
+-}
+blockMember : String -> JsonValue -> Result String JsonValue
+blockMember email libraryData =
+    Json.Value.setIn [ email ] (BoolValue True) libraryData
```
Comment (viebel) [inline challenges/elm/src/Detailed.elm:113]: I don't understand how it works: you need to access `userManagement` inside `libraryData`!

File: challenges/elm/src/Minimal.elm (hunk 1 above)
Comment (wolfadex) [reply challenges/elm/src/Minimal.elm:26]: Totally could do it that way too. I could update it

File: challenges/elm/src/Detailed.elm (hunk 3 above)
Comment (wolfadex) [reply challenges/elm/src/Detailed.elm:113]: I think I made a mistake here, I'll update it

File: challenges/elm/src/Minimal.elm (hunk 1 above)
Comment (viebel) [reply challenges/elm/src/Minimal.elm:26]: Please do. I'd like to see how we manipulate a json in a statically typed language!

**Comment by viebel** [comment]
Please send me a message on Twitter to claim your free copy of the book!

**Comment by wolfadex** [comment]
> Please send me a message on Twitter to claim your free copy of the book!

Thank you, please offer it to someone else though as I was just happy to contribute sharing knowledge. Especially since there was a discussion on [lobste.rs](https://lobste.rs/) about how this would work in a statically typed language

**Pull request merged**